platform/posix/transport/src/openssl_posix.c

```diff
@@ -792,47 +838,72 @@ int32_t Openssl_Send( NetworkContext_t * pNetworkContext,
 {
     OpensslParams_t * pOpensslParams = NULL;
     int32_t bytesSent = 0;
-    int32_t sslError = 0;
-
-    /* Unused parameter when logs are disabled. */
-    ( void ) sslError;
 
     if( ( pNetworkContext == NULL ) || ( pNetworkContext->pParams == NULL ) )
     {
         LogError( ( "Parameter check failed: pNetworkContext is NULL." ) );
     }
     else if( pNetworkContext->pParams->pSsl != NULL )
     {
+        struct pollfd pollFds;
+        int32_t pollStatus;
+
         pOpensslParams = pNetworkContext->pParams;
-        /* SSL write of data. */
-        bytesSent = ( int32_t ) SSL_write( pOpensslParams->pSsl,
-                                           pBuffer,
-                                           ( int32_t ) bytesToSend );
 
-        if( bytesSent <= 0 )
+        /* Initialize the file descriptor. */
+        pollFds.events = POLLOUT;
+        pollFds.revents = 0;
+        /* Set the file descriptor for poll. */
+        pollFds.fd = pOpensslParams->socketDescriptor;
+
+        /* `poll` checks if the socket is ready to send data. */
```

Comment:
```suggestion
        /* `poll` checks if the socket is ready to send data. 
         * Note: This is done to avoid blocking on SSL_write()
         * when TCP socket is not ready to accept more data for
         * network transmission (possibly due to a full TX buffer). */
```